eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -512,9 +517,9 @@ final case class DATA_NEGOTIATING(commitments: Commitments,
   require(!commitments.localParams.isInitiator || closingTxProposed.forall(_.nonEmpty), "initiator must have at least one closing signature for every negotiation attempt because it initiates the closing")
 }
 final case class DATA_CLOSING(commitments: Commitments,
-                              fundingTx: Option[Transaction], // this will be non-empty if we are the initiator and we got in closing while waiting for our own tx to be published
+                              fundingTx: Option[UnconfirmedFundingTx],
```

Comment:
The new trait is a bit verbose, but it highlights the fact that this is only used when the funding tx is unconfirmed (which wasn't obvious before). Alternatively, we could just use an `Option[Either[Transaction, SignedSharedTransaction]]`, let me know what you prefer.